### PR TITLE
dev/sg: implement 'sg ops update-images -k compose'

### PR DESCRIPTION
**Pull request by bobheadxi**
Allows `sg ops update-images` to work on Compose manifests.

From a hack session with @davejrt where we tried to deploy otel-collector in a compose deployment - to try opentelemetry we needed the latest insiders images.

## Test plan

```
go build -o ../deploy-sourcegraph-docker/sg ./dev/sg
```

then in `deploy-sourcegraph-docker`:

```
./sg -v ops update-images -k compose -t insiders ./docker-compose
```

### CODE REVIEW

File: dev/sg/internal/images/compose.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	services, ok := compose["services"].(map[string]any)
+	if !ok {
+		return nil, errors.New("invalid services")
+	}
+
+	type replace struct {
+		original string
+		new      string
+	}
+	checks := group.NewWithResults[*replace]().WithConcurrencyLimiter(group.NewBasicLimiter(10))
```
Comment (michaellzc) [inline dev/sg/internal/images/compose.go:79]: 🤩 ahh nice, this would be a lot faster than the other deployment type cuz they're done sequentially

**Review by michaellzc — APPROVED** [review]

File: dev/sg/internal/images/compose.go (hunk 1 above)
Comment (bobheadxi) [reply dev/sg/internal/images/compose.go:79]: Yes, if I have time this week (maybe hack hour?) I was thinking of cleaning up the other implementations, having them all check for updates concurrently, and add a flag to configure the concurrency

Comment (michaellzc) [reply dev/sg/internal/images/compose.go:79]: > Yes, if I have time this week (maybe hack hour?) I was thinking of cleaning up the other implementations, having them all check for updates concurrently, and add a flag to configure the concurrency

oh yes please fix my naive go code

Comment (bobheadxi) [reply dev/sg/internal/images/compose.go:79]: also thanks again @camdencheek for this beautiful library 😍

Comment (camdencheek) [inline dev/sg/internal/images/compose.go:79]: Glad you're enjoying it 😄 

FYI, it also provides a shortcut for creating a group with a limiter

```suggestion
	checks := group.NewWithResults[*replace]().WithMaxConcurrency(10)
```

**Review by kalanchan — APPROVED** [review]

**Pull request merged**